aten/src/ATen/core/UndefinedTensorImpl.cpp

```diff
@@ -35,6 +35,10 @@ int64_t UndefinedTensorImpl::storage_offset() const {
 IntList UndefinedTensorImpl::strides() const {
   AT_ERROR("strides() called on undefined Tensor");
 }
+
+bool UndefinedTensorImpl::defined() const {
```

Comment:
as above, I don't really understand why we need this -- having a Variable with UndefinedTensorImpl seems bogus.